core/src/main/java/org/apache/calcite/sql/SqlDataTypeSpec.java

```diff
@@ -153,13 +167,42 @@ public Boolean getNullable() {
     return nullable;
   }
 
+  public Boolean getUppercase() {
+    return uppercase;
+  }
+
+  public Boolean getCaseSpecific() {
+    return caseSpecific;
+  }
+
   /** Returns a copy of this data type specification with a given
    * nullability. */
   public SqlDataTypeSpec withNullable(Boolean nullable) {
     if (Objects.equals(nullable, this.nullable)) {
       return this;
     }
-    return new SqlDataTypeSpec(typeNameSpec, timeZone, nullable, getParserPosition());
+    return new SqlDataTypeSpec(typeNameSpec, timeZone, nullable, uppercase,
+        caseSpecific, getParserPosition());
+  }
+
+  /** Returns a copy of this data type specification with a given
+   * uppercase status. */
+  public SqlDataTypeSpec withUppercase(Boolean uppercase) {
+    if (Objects.equals(uppercase, this.uppercase)) {
+      return this;
+    }
+    return new SqlDataTypeSpec(typeNameSpec, timeZone, nullable, uppercase,
+        caseSpecific, getParserPosition());
+  }
+
+  /** Returns a copy of this data type specification with a given
+   * nullability. */
```

Comment:
I think this comment could be updated for case specific instead of nullability

Reply:
You're correct, darn copy and paste!

Reply:
Also, the indentation looks a little off.